src/main/java/org/reaktivity/nukleus/ws/internal/routable/stream/TargetOutputEstablishedStreamFactory.java

```diff
@@ -324,13 +328,30 @@ private void handleThrottle(
             }
         }
 
-        private void processWindow(
-            DirectBuffer buffer,
-            int index,
-            int length)
+        public void processInitialWindow(WindowFW window)
         {
-            final WindowFW window = windowRO.wrap(buffer, index, index + length);
+            final int sourceWindowBytesDelta = window.update();
+            final int sourceWindowFramesDelta = window.frames();
+
+            final int targetWindowBytesDelta = sourceWindowBytesDelta + targetWindowBytesAdjustment;
+            final int targetWindowFramesDelta = sourceWindowFramesDelta + targetWindowFramesAdjustment;
+
+            targetWindowBytes += Math.max(targetWindowBytesDelta, 0);
+            targetWindowBytesAdjustment = Math.abs(Math.min(targetWindowBytesDelta, 0));
+            targetWindowBytesAdjustment -= sourceWindowBytesDelta * 20 / 100;
+
+            targetWindowFrames += Math.max(targetWindowFramesDelta, 0);
+            targetWindowFramesAdjustment = Math.abs(Math.min(targetWindowFramesDelta, 0));
 
+            if (targetWindowBytesDelta > 0 || targetWindowFramesDelta > 0)
+            {
+                source.doWindow(sourceId, targetWindowBytesDelta, Math.max(targetWindowFramesDelta, 0));
+                this.windowHandler = this::processWindow;
+            }
+        }
```

Comment:
```
public void processInitialWindow(WindowFW window)
{
    final int sourceWindowBytesDelta = window.update();
    targetWindowBytesAdjustment -= sourceWindowBytesDelta * 20 / 100;
    windowHandler = this::processWindow;
    windowHandler.accept(window)
}
```
I was thinking to avoid the code duplication, more like above - does that work?